packages/api/source/controllers/MembershipController.ts

```diff
@@ -0,0 +1,155 @@
+        status,
+        createdAt,
+        updatedAt,
+    } = membership;
+    return {
+        id: id || _id.toString(),
+        member,
+        inviter,
+        division,
+        type,
+        status,
+        createdAt,
+        updatedAt,
+    };
+};
+
+const create = async (context, attributes): Promise<ExternalMembership> => {
+    const { error, value } = createSchema.validate(attributes, {
+        stripUnknown: true,
+    });
+
+    if (error) {
+        throw new BadRequestError(error.message);
+    }
+
+    const { emailAddress, organizationId } = value;
+    let member = await UserModel.findOne({ emailAddress });
+
+    if (!member) {
+        const newUser = new UserModel({
+            firstName: "<unavailable>",
+            lastName: "<unavailable>",
+            description: "<unavailable>",
+            gender: undefined,
+            countryCode: undefined,
+            emailAddress: emailAddress,
+            emailVerified: true,
+            status: "invited",
+            role: "viewer",
+        });
+
+        await newUser.save();
+        member = newUser;
+    }
+
+    let membership = await MembershipModel.findOne({
+        member: member.id,
+        organizationId,
+    });
+
+    if (!membership) {
+        const newMembership = new MembershipModel({
+            member: member.id,
+            inviter: "61fcf3cd1ca8c8033509f728",
+            division: organizationId,
+            type: "organization",
+            status: "invited",
+        });
+        await newMembership.save();
+        membership = newMembership;
+    }
+
+    if (membership.status === "invited") {
+        const token = jwt.sign(
+            { emailAddress, organizationId },
+            INVITATION_JWT_SIGNATURE,
+            {
+                expiresIn: 60 * 60,
+            },
+        );
+        const subject = "";
+        const text = "";
+        const html = invitationTemplate(token);
+        sendEmail(emailAddress, subject, text, html);
+    }
+
+    return toExternal(membership);
+};
+
+const verify = async (context, token): Promise<Boolean> => {
+    try {
+        const { emailAddress, organizationId } = jwt.verify(
+            token,
+            INVITATION_JWT_SIGNATURE,
+        );
+
+        await runAsTransaction(async () => {
+            const user = await UserModel.findOneAndUpdate(
+                { emailAddress },
+                {
+                    $set: { organization: organizationId },
+                },
+                { new: true },
+            );
+
+            await MembershipModel.findOneAndUpdate(
+                {
+                    member: user.id,
+                    organizationId,
+                },
+                {
+                    status: "accepted",
+                },
+            );
+
+            await OrganizationModel.findByIdAndUpdate(
+                organizationId,
+                { $push: { members: user.id } },
+                { safe: true, upsert: true },
```

Comment:
Why are we upserting? An organization should exist if the user should be invited to it!